libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3121,6 +3120,11 @@ string YulUtilFunctions::validatorFunction(Type const& _type, bool _revertOnFail
 			solAssert(false, "Validation of type " + _type.identifier() + " requested.");
 		}
 
+		if (_revertOnFailure)
+			templ("failure", "revert(0, 0)");
+		else
+			templ("failure", panicFunction(panicCode) + "()");
```

Comment:
What exactly would trigger this? ABI decoding for example?

Reply:
No, abi decoding failure results in revert. `_revertOnFailure` is currently only false for enum conversion checks.